File: app/models/shoe.rb
class Shoe < ApplicationRecord
  belongs_to :user

  validates :model, presence: true
  validates :version, presence: true
  validates :brand, presence: true
  validates :picture, presence: true
  validates :year, presence: true, length: { is: 4 }, numericality: { only_integer: true }
  validates :price, presence: true
  validates :price, format: { with: /\A\d+(?:\.\d{0,2})?\z/ }, numericality: {greater_than: 0}
  validates :description, presence: true
  validates :user, presence: true
end


# validates :picture, inclusion: { in: %w(http),
# message: "%{value} is not a valid link.  Please include an http link." }

File: react/src/components/Shoe.js
import React from 'react';

const Shoe = (props) => {
  return(
    <div>
      <h2><a href={'/shoes/' + props.id}>{props.brand} {props.model} {props.version}</a></h2>
      <img src={`${props.picture}`}/>
      <h4>{props.description} {props.year}</h4>
      <p>{props.price}</p>
    </div>
  )

}

export default Shoe;